### ISSUE TITLE
Flag the user if the piggy bank being linked to a transaction does not draw from the destination account

### ISSUE THREAD

**Issue by isaacgr** [firefly-iii/firefly-iii#9312]
### Support guidelines

- [x] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [x] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [ ] ... there are no open or closed issues that are related to my problem
- [ ] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

v6.1.16

I had entered a transaction for a transfer from chequing to savings. I selected the piggy bank I wanted to associate it with, however noticed that the piggy bank never got updated.

This is really 2 issues:

1. Editing the transaction, there was no way for me to change the piggy bank. I had to delete and re-create the transfer to select the proper piggy bank
2. The piggy bank I was selecting didnt draw from the transfer account, so it took me a bit to realize that was why it didnt show up at all

This might partially be fixed with #6557, though I'm not sure if it will include historical changes (i.e. if I do the above and then later associate the piggy bank also with that account).

### Debug information

- Tried editing the transfer, but it doesn't appear to have the option to modify the piggy bank on either edit page
- deleted and re-added

### Expected behaviour

- flag the user if the piggy bank doesn't draw from the destination account?
- it sounds like the ability to edit after the fact wont be added (though this doesnt quite say why) [#9203](https://github.com/firefly-iii/firefly-iii/issues/9203)

**Comment by JC5** [comment]
Thanks for opening an issue!

You don't need to recreate a transaction to withdraw money from a piggy bank. You can do it manually. 

So no, this will not be a feature. The ability to select a piggy bank is just a quick shortcut. You can always do it by hand if you happen to make a mistake.

**Comment by isaacgr** [comment]
But if it’s a shortcut then should it matter if it’s not drawing from that account (and so just apply the amount regardless)? The issue is more visibility than anything. This behaviour isn’t 100% clear and took me a bit to figure out really what was wrong. 

Any reason for not allowing it to be edited after the fact? Are they just not actually linked I guess?

**Comment by JC5** [comment]
Indeed, they are not linked. But I'll write it down as an enhancement, I'll think of something :)
